sonic_platform_base/sonic_sfp/sfputilhelper.py

```diff
@@ -90,7 +90,7 @@ def read_porttab_mappings(self, porttabfile, asic_inst=0):
             self.logical_to_asic[intf_name] = asic_inst
 
         self.logical.extend(logical)
-        self.logical = list(set(self.logical))
+        self.logical = list(OrderedDict.fromkeys(self.logical).keys())
```

Comment:
@sheiun-xu can you explain why natsorted() at line 76 on logical port is not working?

Reply:
@prgeor `natsorted()` at line 76 did work but after that the `set(self.logical)` broke the order.

Ex.

```python
>>> from natsort import natsorted
>>> intfs = natsorted([f"Ethernet{i}" for i in range(21)], lambda y: y.lower())
>>> print(intfs)
['Ethernet0', 'Ethernet1', 'Ethernet2', 'Ethernet3', 'Ethernet4', 'Ethernet5', 'Ethernet6', 'Ethernet7', 'Ethernet8', 'Ethernet9', 'Ethernet10', 'Ethernet11', 'Ethernet12', 'Ethernet13', 'Ethernet14', 'Ethernet15', 'Ethernet16', 'Ethernet17', 'Ethernet18', 'Ethernet19', 'Ethernet20']
>>> print(list(set(intfs)))
['Ethernet8', 'Ethernet15', 'Ethernet2', 'Ethernet7', 'Ethernet9', 'Ethernet1', 'Ethernet14', 'Ethernet16', 'Ethernet19', 'Ethernet18', 'Ethernet10', 'Ethernet17', 'Ethernet5', 'Ethernet11', 'Ethernet4', 'Ethernet3', 'Ethernet0', 'Ethernet12', 'Ethernet6', 'Ethernet13']
```